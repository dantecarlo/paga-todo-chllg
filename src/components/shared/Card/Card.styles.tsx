import styled from 'styled-components'

export const StyledCardContainer = styled('div')`
  display: flex;
  flex-direction: column;
  width: 23rem;
  height: 12rem;
  background: rgba(255, 255, 255, 0.15);
  border-radius: 16px;
  box-shadow: 0 4px 30px rgba(0, 0, 0, 0.1);
  backdrop-filter: blur(6.2px);
  -webkit-backdrop-filter: blur(6.2px);
  border: 1px solid rgba(255, 255, 255, 0.1);
  padding: 2rem;

  @media (max-width: 1024px) {
    width: 80%;
  }
`

export const StyledCardHeader = styled('div')`
  display: flex;
  justify-content: space-between;
  align-items: center;
`

export const StyledCardHeaderText = styled('div')`
  display: flex;
  flex-direction: column;
`

export const StyledImage = styled('img')`
  width: 5rem;
  height: 5rem;
  border-radius: 50%;
  object-fit: cover;
`

export const StyledCardH2 = styled('h2')`
  font-weight: 600;
  color: var(--color-font);
  margin: 0;
  margin-bottom: 2rem;
`

export const StyledCardH3 = styled('h3')`
  font-weight: 400;
  color: var(--color-font);
  margin: 0;
  margin-bottom: 0.5rem;
`

export const StyledCardH4 = styled('h4')`
  font-weight: 600;
  color: var(--color-font);
  margin: 0;
  margin-bottom: 0.1rem;
`

export const StyledCardH5 = styled('h5')`
  font-weight: 400;
  color: var(--color-font);
  margin: 0;
  margin-bottom: 1rem;

  &:last-child {
    margin-bottom: 0;
  }
`
