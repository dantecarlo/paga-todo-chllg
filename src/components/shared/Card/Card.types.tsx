export interface ICardInterface {
  bankName: string
  age: number
  description: string
  url: string
  loading: boolean
  favorite?: boolean
  onClick?: () => void
}
