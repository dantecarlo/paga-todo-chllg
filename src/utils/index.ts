export * from './constants/i18n.constants'
export * from './constants/layout.constants'
export * from './constants/navigation.constants'
export * from './testUtils'
