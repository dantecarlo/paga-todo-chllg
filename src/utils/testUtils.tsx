import { render } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'

const baseWrapper = (componentTree: JSX.Element) => {
  return componentTree
}

const wrapInBrowser = (componentTree: JSX.Element) => {
  return <BrowserRouter>{componentTree}</BrowserRouter>
}

const setupComponent = (ui: JSX.Element) => {
  let componentTree = baseWrapper(ui)

  componentTree = wrapInBrowser(componentTree)

  return componentTree
}

const customRender = (ui: JSX.Element) => {
  const componentTree = setupComponent(ui)
  return render(componentTree)
}

export * from '@testing-library/react'

export { customRender as render }
