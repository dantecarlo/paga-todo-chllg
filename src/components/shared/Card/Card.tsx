import { faHeart as faHeartRegular } from '@fortawesome/free-regular-svg-icons'
import { faHeart } from '@fortawesome/free-solid-svg-icons'
import { FC } from 'react'
import { useTranslation } from 'react-i18next'

import LoadingSpinner from '../LoadingSpinner'
import {
  StyledCardContainer,
  StyledCardH2,
  StyledCardH3,
  StyledCardH4,
  StyledCardH5,
  StyledCardHeader,
  StyledCardHeaderText,
  StyledHeartIcon,
  StyledImage
} from './Card.styles'
import { ICardInterface } from './Card.types'

const Card: FC<ICardInterface> = ({
  bankName = '',
  age = -1,
  description = '',
  url = '',
  loading = false,
  favorite = false,
  onClick = () => {}
}) => {
  const { t } = useTranslation('global')

  return (
    <StyledCardContainer onClick={onClick}>
      <StyledHeartIcon icon={favorite ? faHeart : faHeartRegular} />

      {loading ? (
        <LoadingSpinner />
      ) : (
        <>
          <StyledCardHeader>
            <StyledCardHeaderText>
              <StyledCardH3>{t('Component.Card.Bank') as string}</StyledCardH3>
              <StyledCardH2>{bankName}</StyledCardH2>
            </StyledCardHeaderText>
            <StyledImage loading="lazy" src={url} />
          </StyledCardHeader>
          <StyledCardH4>{t('Component.Card.Age') as string}</StyledCardH4>
          <StyledCardH5>{age}</StyledCardH5>
          <StyledCardH4>
            {t('Component.Card.Description') as string}
          </StyledCardH4>
          <StyledCardH5>{description}</StyledCardH5>
        </>
      )}
    </StyledCardContainer>
  )
}

export default Card
