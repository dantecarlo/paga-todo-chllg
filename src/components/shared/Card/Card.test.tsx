import { cleanup, render, RenderResult } from 'utils/testUtils'

import Card from './Card'

const bankData = {
  bankName: 'Paga Todo',
  description: 'Banco Paga Todo es Para Todos',
  age: 10,
  url: 'http://www.one.url.com'
}

describe('<Card />', () => {
  let component: RenderResult

  beforeEach(() => {
    component = render(
      <Card
        age={bankData.age}
        bankName={bankData.bankName}
        description={bankData.description}
        loading={false}
        url={bankData.url}
      />
    )
  })

  afterEach(() => {
    cleanup()
  })

  test('render correctly', () => {
    expect(component).toMatchSnapshot()
  })
})
