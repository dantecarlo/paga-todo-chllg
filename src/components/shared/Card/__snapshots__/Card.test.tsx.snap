// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gswNZR fVFpTZ"
      >
        <div
          class="sc-dkrFOg huHyJh"
        >
          <div
            class="sc-hLBbgP hzsEKv"
          >
            <h3
              class="sc-gKPRtg cChpIe"
            >
              Component.Card.Bank
            </h3>
            <h2
              class="sc-jSUZER htltFF"
            >
              Paga Todo
            </h2>
          </div>
          <img
            class="sc-eDvSVe fydxnx"
            loading="lazy"
            src="http://www.one.url.com"
          />
        </div>
        <h4
          class="sc-iBYQkv cLHOkr"
        >
          Component.Card.Age
        </h4>
        <h5
          class="sc-ftTHYK dCZyn"
        >
          10
        </h5>
        <h4
          class="sc-iBYQkv cLHOkr"
        >
          Component.Card.Description
        </h4>
        <h5
          class="sc-ftTHYK dCZyn"
        >
          Banco Paga Todo es Para Todos
        </h5>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gswNZR fVFpTZ"
    >
      <div
        class="sc-dkrFOg huHyJh"
      >
        <div
          class="sc-hLBbgP hzsEKv"
        >
          <h3
            class="sc-gKPRtg cChpIe"
          >
            Component.Card.Bank
          </h3>
          <h2
            class="sc-jSUZER htltFF"
          >
            Paga Todo
          </h2>
        </div>
        <img
          class="sc-eDvSVe fydxnx"
          loading="lazy"
          src="http://www.one.url.com"
        />
      </div>
      <h4
        class="sc-iBYQkv cLHOkr"
      >
        Component.Card.Age
      </h4>
      <h5
        class="sc-ftTHYK dCZyn"
      >
        10
      </h5>
      <h4
        class="sc-iBYQkv cLHOkr"
      >
        Component.Card.Description
      </h4>
      <h5
        class="sc-ftTHYK dCZyn"
      >
        Banco Paga Todo es Para Todos
      </h5>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
